### PR TITLE
Fix application error

### PR DESCRIPTION
**Pull request by gomorizsolt**
Issue #128.

TODOs once it's approved:
- update config
- deploy to https://github.com/c-hive/c-hive-dotdev

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -1,18 +1,36 @@
...
-  </ThemeProvider>
+const indexPage = ({ config }) => (
+  <>
+    <Head>
+      <title>title</title>
+    </Head>
+    <ThemeProvider>
+      <Normalize />
+      <GlobalStyle />
+      <ConfigProvider config={config}>
```
Comment (gomorizsolt) [inline pages/index.js:18]: Perhaps it's weird to pass something to a context provider but it's required as `getStaticProps()` is limited for pages: https://nextjs.org/docs/basic-features/data-fetching#only-allowed-in-a-page (also left this as a comment below).

**Comment by thisismydesign** [comment]
> I couldn't override the NODE_ENV environment variable for some reason

In which environment? If there are any defaults wouldn't they work?

**Comment by gomorizsolt** [comment]
As discussed over the phone, let's get rid of the `TARGET` environment variable (reason: it's just an additional layer of complexity).

**Comment by gomorizsolt** [comment]
I've tested these changes against my forked version, works pretty well! 👍

File: README.md [hunk 2]
```diff
@@ -20,11 +20,11 @@ Refer to the [example](./config/config.example.yml) file for the available optio
 #### Deploy to GitHub Pages
 - [Fork me](/../../fork)
 - Store a [deployment key](https://github.com/JamesIves/github-pages-deploy-action/tree/dev#using-an-ssh-deploy-key-) in the `DEPLOY_KEY` secret
-- Store config in `CONFIG` secret
+- Store the output of `npm run print-config` in the `CONFIG` environment variable
```
Comment (thisismydesign) [inline README.md:23]:
```suggestion
- Edit `config/config.yml`
- Run `npm run print-config`
- Store the output in the `CONFIG` secret
```

File: README.md [hunk 3]
```diff
@@ -20,11 +20,11 @@ Refer to the [example](./config/config.example.yml) file for the available optio
...
 #### Deploy elsewhere
 - [Fork me](/../../fork)
 - Set up custom deployment workflow
-- Store the configuration in the `CONFIG` environment variable
+- Store the output of `npm run print-config` in the `CONFIG` environment variable
```
Comment (thisismydesign) [inline README.md:27]:
```suggestion
- Edit `config/config.yml`
- Run `npm run print-config`
- Store the output in the `CONFIG` environment variable
```

File: scripts/read-config.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import path from "path";
+import fs from "fs";
+import yaml from "js-yaml";
+
+const readConfig = () => {
+  if (process.env.NODE_ENV === "production") {
+    if (!process.env.CONFIG) {
```
Comment (thisismydesign) [inline scripts/read-config.js:7]: Let's refactor this into `readProductionConfig` and `readDevelopmentConfig` methods.

```
if (process.env.NODE_ENV === "production")
  readProductionConfig()
else
  readDevelopmentConfig()
end
```

File: README.md (hunk 2 above)
Comment (gomorizsolt) [reply README.md]: https://github.com/c-hive/dotdev/pull/138/commits/7f816ffe4f6a9fadf9172204898857e9e84b60d4

File: README.md (hunk 3 above)
Comment (gomorizsolt) [reply README.md]: https://github.com/c-hive/dotdev/pull/138/commits/c987cbf9b1effd083ba582a53d1f5ded763b996e

File: scripts/read-config.js (hunk 4 above)
Comment (gomorizsolt) [reply scripts/read-config.js:7]: Thanks for pointing this out! Yea, that nested `if()` statement was a bit confusing to me as well.

Comment (gomorizsolt) [reply scripts/read-config.js]: https://github.com/c-hive/dotdev/pull/138/commits/649b38d965aa3b8183721c14e5e9652ab67305eb

File: scripts/read-config.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import path from "path";
+import fs from "fs";
+import yaml from "js-yaml";
+
+const readProductionConfig = () => {
+  if (!process.env.CONFIG) {
+    throw new Error(
+      "Cannot find `CONFIG` environment variable. Make sure it's configured and exposed."
```
Comment (thisismydesign) [inline scripts/read-config.js:8]:
```suggestion
      "`CONFIG` environment variable is not set."
```

Comment (gomorizsolt) [reply scripts/read-config.js]: https://github.com/c-hive/dotdev/pull/138/commits/1b4b5dfb1ed6c47618b1777e4dcaea51468046a6

**Pull request merged**